### PR TITLE
[LDAF-427] Update colors across the site (links, buttons, etc.)

### PR DESCRIPTION
**Pull request by LouisFettet**
Jira ticket: [LDAF-427](https://ldaf.atlassian.net/browse/LDAF-427)

The Figma mocks for [links](https://www.figma.com/file/oGKbyCnCRRdNzLYbiags93/LDAF-Component-Library-USWDS-3.0.2?type=design&node-id=2178-9666&mode=design&t=cey80GrlG7U7XGB3-4) and [buttons](https://www.figma.com/file/oGKbyCnCRRdNzLYbiags93/LDAF-Component-Library-USWDS-3.0.2?type=design&mode=design&t=cey80GrlG7U7XGB3-0) have been updated after an audit of our colors. This might not cover everything, but it will at least get us closer to the mocks.

## Proposed changes

- Make a handful of color updates based on mocks.
- Utilize the `<Link/>` component wherever it makes sense to do so.
- Override the `usa-link` class colors provided by the USWDS.
- Utilize our color variables in the `<Button/>` component styling, and ensure links that render with button styling utilize the button styling (e.g. no `visited` state). Add a new story for every button variation when it is rendered as link.
- Remove the "secondary" button variation from use. CTA buttons on the homepage, top tier pages, and core content pages should all now show the first CTA with the "primary" variation and all following CTAs with the "outline" variation.
- Provide some default `focus` styles to override the USWDS default outline color. Make a few updates so that focus styles on dark backgrounds use a lighter outline.
- Handle a metric ton of edge cases for setting colors. These are mostly for situations where we need to override the USWDS when it's being overly specific. Some examples include:
  -  the "Return to top" link in the footer 
  - breadcrumb links
  - skip nav link
  - primary and secondary nav links
  - sidenav links
  - special overrides for `focus` outlines on elements in the header at different screen sizes
  - video card links

## Acceptance criteria validation

- [x] Ensure all links are using the `<Link/>` component as appropriate.
- [x] All links (except special cases such as the secondary nav in the header and the links in the bottom half of the footer) should be using the colors outlined on the [typography board in Figma](https://www.figma.com/file/oGKbyCnCRRdNzLYbiags93/LDAF-Component-Library-USWDS-3.0.2?type=design&node-id=2178-9593&mode=design&t=i8ySkXmzN7RwXtom-4).

### Alternate solutions

Ideally we wouldn't need to override the USWDS classes, but this doesn't seem like an option given our color palette. See [my message about this in Slack](https://adhoc.slack.com/archives/C04EYUA76BT/p1695229935249559) for more information.

## Requested feedback

See self-review. This is all way more brittle than I'd like, so any and all feedback is welcome!

[LDAF-427]: https://ldaf.atlassian.net/browse/LDAF-427?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/components/Button/Button.scss [hunk 1]
```diff
@@ -55,116 +42,137 @@ $filters: (
...
   }
   &:hover {
-    background: map.get($colors, "primary-dark");
+    color: #fff;
+    background: $primary-dark;
   }
   &:active {
-    background: map.get($colors, "primary-darker");
+    color: #fff;
+    background: $primary-darkest;
```
Comment (LouisFettet) [inline src/lib/components/Button/Button.scss:63]: `primary-darker` in the map is the same as the `$primary-darkest` variable (`#00284d`).

File: src/lib/components/Button/Button.scss [hunk 2]
```diff
@@ -55,116 +42,137 @@ $filters: (
...
+  }
+  &:visited {
+    color: #fff;
   }
   &:disabled,
   &:disabled:hover,
   &:disabled:active {
-    background: map.get($colors, "disabled-dark");
+    background: $disabled-darker;
```
Comment (LouisFettet) [inline src/lib/components/Button/Button.scss:71]: `disabled-dark` in the map is the same as the `$disabled-darker` variable (`#adadad`).

File: src/lib/components/Button/Button.scss [hunk 3]
```diff
@@ -55,116 +42,137 @@ $filters: (
...
   }
+  &:visited {
+    color: $primary;
+  }
   &:disabled,
   &:disabled:hover,
   &:disabled:active {
     background: none;
-    color: map.get($colors, "disabled");
+    color: $disabled-dark;
```
Comment (LouisFettet) [inline src/lib/components/Button/Button.scss:175]: `disabled` in the map is the same as the `$disabled-dark` variable (`#c9c9c9`).

**Review by LouisFettet — COMMENTED** [review]
Some quick notes on color updates for the `<Button/>` component. Everything else should be documented with in-code comments.

**Review by Benaiah — APPROVED** [review]

**Comment by LouisFettet** [comment]
Closing in favor of #489!

**Pull request closed without merging**